geonode/layers/views.py

```diff
@@ -313,6 +313,8 @@ def layer_style_upload(request):
         sld = request.FILES['sld_file'].read()
 
         set_layer_style(layer, data.get('layer_title'), sld)
+        from geonode.geoserver.tasks import geoserver_create_thumbnail
+        geoserver_create_thumbnail.apply_async(((layer.resourcebase_ptr.id, True, True)))
```

Comment:
By policy we don't want the thumbnails automatically overridden by some GeoNode view. Better to remove those lines here.